internal/internal_event_handlers.go

```diff
@@ -818,3 +858,30 @@ func (weh *workflowExecutionEventHandlerImpl) handleChildWorkflowExecutionTermin
 
 	return nil
 }
+
+func (weh *workflowExecutionEventHandlerImpl) handleSignalExternalWorkflowExecutionRequested(event *m.HistoryEvent) error {
+	attributes := event.ExternalWorkflowExecutionSignalRequestedEventAttributes
+	signalID := string(attributes.Control)
```

Comment:
the name "Control" is confusing. If it is used as signalID, name it as signalID

Reply:
I did a short discuss with Maxim and Samar, we are unsure about using Control or SignalID. Since everywhere in client is using Control, will keep it like this.